rust/core-lib/src/view.rs

```diff
@@ -882,6 +883,19 @@ impl View {
             self.set_selection(text, occ);
         }
     }
+
+    pub fn get_line_range(&self, text: &Rope, region: &SelRegion) -> Range<usize> {
+        let (first_line, _) = self.offset_to_line_col(text, region.min());
+        let (last_line, last_col) =
+            self.offset_to_line_col(text, region.max());
+        let last_line = if last_col == 0 && last_line > first_line {
```

Comment:
Just as point of style, this reads cleaner to me to make `last_line` mut and just do `-= 1` if the condition holds.